pkg/node/node.go

```diff
@@ -278,37 +278,58 @@ func NewBee(
 		return nil, fmt.Errorf("compute overlay address: %w", err)
 	}
 
-	if nonceExists && o.TargetNeighborhood != "" {
-		logger.Warning("an overlay has already been created before, skipping targeting the selected neighborhood")
+	targetNeighborhood := o.TargetNeighborhood
+	if targetNeighborhood == "" && !nonceExists && o.NeighborhoodSuggester != "" {
+		logger.Info("fetching target neighborhood from suggester", "url", o.NeighborhoodSuggester)
+		targetNeighborhood, err = nbhdutil.FetchNeighborhood(&http.Client{}, o.NeighborhoodSuggester)
+		if err != nil {
+			return nil, fmt.Errorf("neighborhood suggestion: %w", err)
+		}
 	}
 
-	if !nonceExists {
-		// mine the overlay
-		targetNeighborhood := o.TargetNeighborhood
-		if o.TargetNeighborhood == "" && o.NeighborhoodSuggester != "" {
-			logger.Info("fetching target neighborhood from suggester", "url", o.NeighborhoodSuggester)
-			targetNeighborhood, err = nbhdutil.FetchNeighborhood(&http.Client{}, o.NeighborhoodSuggester)
-			if err != nil {
-				return nil, fmt.Errorf("neighborhood suggestion: %w", err)
-			}
+	if targetNeighborhood != "" {
+		neighborhood, err := swarm.ParseBitStrAddress(targetNeighborhood)
+		if err != nil {
+			return nil, fmt.Errorf("invalid neighborhood. %s", targetNeighborhood)
 		}
 
-		if targetNeighborhood != "" {
+		if swarm.Proximity(swarmAddress.Bytes(), neighborhood.Bytes()) < uint8(len(targetNeighborhood)) {
+			if nonceExists {
```

Comment:
Removing directory before the `nbhdutil.MineOverlay`, which can fail, may result with only removed data.